File: clients/apple/Packages/MailcalKit/Sources/MailcalUI/NotificationPrefs.swift
// The user's "new-mail notifications" on/off choice, stored in UserDefaults. A client-side
// preference (the background sync still runs and advances the core's marks when off, only
// *posting* a notification is gated, so toggling off then on never floods with a backlog).
import Foundation

enum NotificationPrefs {
    private static let key = "new_mail_notifications_enabled"

    /// Whether new-mail notifications are on. Defaults to on for a fresh install.
    static var enabled: Bool {
        get {
            // Absent key ⇒ default on (a fresh install opts in until the user turns it off).
            guard AppPrefs.defaults.object(forKey: key) != nil else { return true }
            return AppPrefs.defaults.bool(forKey: key)
        }
        set { AppPrefs.defaults.set(newValue, forKey: key) }
    }
}

File: clients/android/app/src/main/java/eu/allodia/mailcal/NotificationPrefs.kt
// The user's "new-mail notifications" on/off choice, a small client-side preference (the core
// still runs the background sync and advances its marks regardless; this only gates whether the
// worker *posts* a notification, so toggling off then on never floods with a backlog).
package eu.allodia.mailcal

import android.content.Context

object NotificationPrefs {
    private const val FILE = "mailcal_prefs"
    private const val KEY_ENABLED = "new_mail_notifications_enabled"

    /// Whether new-mail notifications are on (the default for a fresh install).
    fun enabled(context: Context): Boolean =
        prefs(context).getBoolean(KEY_ENABLED, true)

    /// Persists the user's choice.
    fun setEnabled(context: Context, value: Boolean) {
        prefs(context).edit().putBoolean(KEY_ENABLED, value).apply()
    }

    private fun prefs(context: Context) =
        context.getSharedPreferences(FILE, Context.MODE_PRIVATE)
}
